src/experimental-layers/src/advanced-text-layer/advanced-text-layer.js

```diff
@@ -0,0 +1,182 @@
+// Copyright (c) 2015 - 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import {CompositeLayer} from 'deck.gl';
+import TextMultiIconLayer from './text-multi-icon-layer';
+
+const DEFAULT_COLOR = [0, 0, 0, 255];
+const TEXT_ANCHOR = {
+  start: 1,
+  middle: 0,
+  end: -1
+};
+const ALIGNMENT_BASELINE = {
+  top: 1,
+  center: 0,
+  bottom: -1
+};
+
+const defaultProps = {
+  getText: x => x.text,
+  getPosition: x => x.coordinates,
+  getColor: x => x.color || DEFAULT_COLOR,
+  getSize: x => x.size || 32,
+  getAngle: x => x.angle || 0,
+  getTextAnchor: x => x.textAnchor || 'middle',
+  getAlignmentBaseline: x => x.alignmentBaseline || 'center',
+  getPixelOffset: x => x.pixelOffset || [0, 0],
+  fp64: false
+};
```

Comment:
declare all props, `fontTexture, fontInfo, ...`. Set them to `null` if no value.

Reply:
OK